src/app/joke/joke.component.ts

```diff
@@ -23,4 +25,19 @@ export class JokeComponent implements OnInit {
       console.log(this.list);
   });
 }
+randomJoke() {
+  this.router.navigate(['jokes/random']);
+  
 }
+
+jokeDetail() {
+// alert('in joke Details()');
+this.apiService.getJokeList().subscribe(data => {
+  this.listArray = data['jokes'];
+  // console.log(this.listArray);
+var rand = this.list[Math.floor(Math.random() * this.listArray.length)];
+alert(this.listArray);
```

Comment:
Lets stop the practice of using alert for debug. console logs are still acceptable.

Reply:
ok, i got it,
i will avoid using alert.